evm/vm/message.py

```diff
@@ -144,6 +144,8 @@ def is_create(self):
 class ShardingMessage(Message):
 
     is_create = False
+    transaction_gas_limit = None
```

Comment:
Thoughts on removing the `transaction_` prefix here?  I'm not sure if my thinking is accurate, but historically, the `Message` object has been intentionally naive about the transaction.  Would it still be accurate if this was simply titled `gas_limit`?

Also, how does do `gas_limit` and `gas` differ?  Now that I'm asking makes me think that this is in-fact appropriately prefixed with `transaction_` assuming that's the difference, one is the message gas, the other is the outer transaction gas limit.

Reply:
Yes, you are correct. One is the outer transaction gas limit and one is message gas. And hence the prefix to tell the difference between them.